x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/bulk_edit_dry_run_confirmation.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { EuiConfirmModal } from '@elastic/eui';
+
+import * as i18n from '../../translations';
+import { DryRunResult } from './use_bulk_actions_dry_run';
+import { BulkEditRuleErrorsList } from './bulk_edit_rule_errors_list';
+
+interface BulkEditDryRunConfirmationProps {
+  result?: DryRunResult;
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+const BulkEditDryRunConfirmationComponent = ({
+  onCancel,
+  onConfirm,
+  result,
+}: BulkEditDryRunConfirmationProps) => {
+  const { failedRulesCount = 0, succeededRulesCount = 0, ruleErrors = [] } = result ?? {};
+  const hasAllSucceeded = failedRulesCount === 0;
+
+  // proceed straight to edit flyout if all rules can be edited successfully
+  useEffect(() => {
+    if (hasAllSucceeded) {
+      setTimeout(onConfirm, 0);
+    }
+  }, [hasAllSucceeded, onConfirm]);
+  if (hasAllSucceeded) {
+    return null;
+  }
```

Comment:
I think in this case it's better to place this logic here.

It's a common practice to return null in render of component if after processing incoming props, component shouldn't be rendered. It allows to keep this logic inside component, rather than expose it outside of the component.